src/main/java/com/pileproject/drivecommand/model/nxt/NxtMachineSimulator.java

```diff
@@ -0,0 +1,52 @@
+package com.pileproject.drivecommand.model.nxt;
+
+import com.pileproject.drivecommand.machine.device.DeviceType;
+import com.pileproject.drivecommand.machine.device.input.LineSensor;
+import com.pileproject.drivecommand.machine.device.input.SoundSensor;
+import com.pileproject.drivecommand.machine.device.input.TouchSensor;
+import com.pileproject.drivecommand.machine.device.input.simulation.SimulatedLineSensor;
+import com.pileproject.drivecommand.machine.device.input.simulation.SimulatedSoundSensor;
+import com.pileproject.drivecommand.machine.device.input.simulation.SimulatedTouchSensor;
+import com.pileproject.drivecommand.machine.device.output.Motor;
+import com.pileproject.drivecommand.machine.device.output.simulation.SimulatedMotor;
+import com.pileproject.drivecommand.machine.device.port.InputPort;
+import com.pileproject.drivecommand.machine.device.port.OutputPort;
+
+public class NxtMachineSimulator extends NxtMachine {
```

Comment:
I'm not familiar with the actual codes, but I think NxtController should take NxtMachine because NxtController binds each port, which is Nxt(Input|Output)Port, with each device like motors and sensors. 
Why is MachineProvider separated with the corresponding controller? In my understanding, the controller class shapes the machine. In other words, User's programs can define the model of his/her machine by implementing the controller class (ex: the machine should have two motors, and two sensors). This means MachineBase itself without controller is useless for the user's code ([drive](https://github.com/PileProject/drive) in our case). That's why I thought "(currently it's NxtController, right?).".